integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/QueryRelationColumnLineageCollector.java

```diff
@@ -0,0 +1,112 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.spark3.agent.lifecycle.plan.column;
+
+import io.openlineage.client.utils.DatasetIdentifier;
+import io.openlineage.spark.agent.lifecycle.plan.column.ColumnLevelLineageContext;
+import io.openlineage.spark.agent.util.ScalaConversionUtils;
+import io.openlineage.sql.ColumnLineage;
+import io.openlineage.sql.ColumnMeta;
+import io.openlineage.sql.OpenLineageSql;
+import io.openlineage.sql.SqlMeta;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.spark.sql.catalyst.expressions.Attribute;
+import org.apache.spark.sql.catalyst.expressions.ExprId;
+import org.apache.spark.sql.catalyst.expressions.NamedExpression;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation;
+import org.apache.spark.sql.execution.datasources.v2.DataSourceV2ScanRelation;
+
+public class QueryRelationColumnLineageCollector {
+
+  public static void extractExternalInputs(ColumnLevelLineageContext context, LogicalPlan plan) {
+    if (plan instanceof DataSourceV2ScanRelation) {
+      extractExternalInputs(context, ((DataSourceV2ScanRelation) plan).relation());
+    }
+    if (plan instanceof DataSourceV2Relation) {
+      extractExternalInputs(context, (DataSourceV2Relation) plan);
+    }
+  }
+
+  public static void extractExternalInputs(
+      ColumnLevelLineageContext context, DataSourceV2Relation relation) {
+    String namespace = relation.table().properties().get("openlineage.dataset.namespace");
```

Comment:
I'm not sure if dataset namespace should be preliminary set in table metadata, or provided as a third argument of this method. For example, all BigQuery datasets have same namespace "bigquery":
https://github.com/OpenLineage/OpenLineage/blob/2d13e38d1a010812df68bd34add239d4af61c6bb/integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/BigQueryNodeInputVisitor.java#L40

Same for query property, it can be extracted from relation in some other way.

Reply:
> How query can be extracted from relation?

E.g. it can be method in catalog which accepts dataset and returns query string.

If this is not some hardcoded value on table level, but a dynamic one, could you please add a comment describing where and how it is implemented? E.g. link to connector documentation, if it is available in public.